# Sendy Logistics Challenge Regression Model¶


 # Introduction #

Sendy is a business-to-business platform established in 2014, to enable businesses of all types and sizes to transport goods more efficiently across East Africa.

This notebook aims to help Sendy predict the estimated time of delivery of orders, from the point of driver pickup to the point of arrival at final destination. This will enhance customer communication and improve the reliability of its service; which will ultimately improve customer experience. In addition, the solution will enable Sendy to realise cost savings, and ultimately reduce the cost of doing business, through improved resource management and planning for order scheduling

An accurate arrival time prediction will help all businesses to improve their logistics and communicate an accurate time to their customers.



# Importing the Libraries

In [4]:
# 
import numpy as np 

# data processing
import pandas as pd 

# data visualization
import seaborn as sns
%matplotlib inline
import matplotlib.pyplot as plt
from matplotlib import style
from matplotlib import rc

# Getting the Data

In [8]:
Test_df= pd.read_csv("Test.csv")
Train_df= pd.read_csv("Train.csv")
VariableDefinitions_df= pd.read_csv("VariableDefinitions.csv")
SampleSubmission_df= pd.read_csv("SampleSubmission.csv")
Riders_df= pd.read_csv("Riders.csv")

# Data Exploration/Analysis

In [9]:
Train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21201 entries, 0 to 21200
Data columns (total 29 columns):
 #   Column                                     Non-Null Count  Dtype  
---  ------                                     --------------  -----  
 0   Order No                                   21201 non-null  object 
 1   User Id                                    21201 non-null  object 
 2   Vehicle Type                               21201 non-null  object 
 3   Platform Type                              21201 non-null  int64  
 4   Personal or Business                       21201 non-null  object 
 5   Placement - Day of Month                   21201 non-null  int64  
 6   Placement - Weekday (Mo = 1)               21201 non-null  int64  
 7   Placement - Time                           21201 non-null  object 
 8   Confirmation - Day of Month                21201 non-null  int64  
 9   Confirmation - Weekday (Mo = 1)            21201 non-null  int64  
 10  Confirmation - Time   

The training-set has 21201 examples and 29 features + the target variable (Time from Pickup to Arrival). 6 of the features are floats, 13 are integers and 10 are objects. Below is the have list of the features with a short description:

In [17]:
Train_df.describe()

Platform Type  Placement - Day of Month  Placement - Weekday (Mo = 1)  \
count   21201.000000              21201.000000                  21201.000000   
mean        2.752182                 15.653696                      3.240083   
std         0.625178                  8.798916                      1.567295   
min         1.000000                  1.000000                      1.000000   
25%         3.000000                  8.000000                      2.000000   
50%         3.000000                 15.000000                      3.000000   
75%         3.000000                 23.000000                      5.000000   
max         4.000000                 31.000000                      7.000000   

       Confirmation - Day of Month  Confirmation - Weekday (Mo = 1)  \
count                 21201.000000                     21201.000000   
mean                     15.653837                         3.240225   
std                       8.798886                         1.567228   
min                       1.000000                         1.000000   
25%                       8.000000                         2.000000   
50%                      15.000000                         3.000000   
75%                      23.000000                         5.000000   
max                      31.000000                         7.000000   

       Arrival at Pickup - Day of Month  Arrival at Pickup - Weekday (Mo = 1)  \
count                      21201.000000                          21201.000000   
mean                          15.653837                              3.240225   
std                            8.798886                              1.567228   
min                            1.000000                              1.000000   
25%                            8.000000                              2.000000   
50%                           15.000000                              3.000000   
75%                           23.000000                              5.000000   
max                           31.000000                              7.000000   

       Pickup - Day of Month  Pickup - Weekday (Mo = 1)  \
count           21201.000000               21201.000000   
mean               15.653837                   3.240225   
std                 8.798886                   1.567228   
min                 1.000000                   1.000000   
25%                 8.000000                   2.000000   
50%                15.000000                   3.000000   
75%                23.000000                   5.000000   
max                31.000000                   7.000000   

       Arrival at Destination - Day of Month  \
count                           21201.000000   
mean                               15.653837   
std                                 8.798886   
min                                 1.000000   
25%                                 8.000000   
50%                                15.000000   
75%                                23.000000   
max                                31.000000   

       Arrival at Destination - Weekday (Mo = 1)  Distance (KM)   Temperature  \
count                               21201.000000   21201.000000  16835.000000   
mean                                    3.240225       9.506533     23.258889   
std                                     1.567228       5.668963      3.615768   
min                                     1.000000       1.000000     11.200000   
25%                                     2.000000       5.000000     20.600000   
50%                                     3.000000       8.000000     23.500000   
75%                                     5.000000      13.000000     26.000000   
max                                     7.000000      49.000000     32.100000   

       Precipitation in millimeters    Pickup Lat   Pickup Long  \
count                    552.000000  21201.000000  21201.000000   
mean                       7.905797     -1.281470     36.811264   
std                 

from the  abobe, it is observable that distance from pickup to drop off location is from 1km to 49km and that on average the Time from Pickup to Arrival is ranging from 1sec 7883 seconds.

In [10]:
Test_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7068 entries, 0 to 7067
Data columns (total 25 columns):
 #   Column                                Non-Null Count  Dtype  
---  ------                                --------------  -----  
 0   Order No                              7068 non-null   object 
 1   User Id                               7068 non-null   object 
 2   Vehicle Type                          7068 non-null   object 
 3   Platform Type                         7068 non-null   int64  
 4   Personal or Business                  7068 non-null   object 
 5   Placement - Day of Month              7068 non-null   int64  
 6   Placement - Weekday (Mo = 1)          7068 non-null   int64  
 7   Placement - Time                      7068 non-null   object 
 8   Confirmation - Day of Month           7068 non-null   int64  
 9   Confirmation - Weekday (Mo = 1)       7068 non-null   int64  
 10  Confirmation - Time                   7068 non-null   object 
 11  Arrival at Pickup

In [11]:
VariableDefinitions_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 2 columns):
 #   Column                               Non-Null Count  Dtype 
---  ------                               --------------  ----- 
 0   Order No                             34 non-null     object
 1   Unique number identifying the order  33 non-null     object
dtypes: object(2)
memory usage: 688.0+ bytes


In [12]:
SampleSubmission_df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7068 entries, 0 to 7067
Data columns (total 2 columns):
 #   Column                       Non-Null Count  Dtype  
---  ------                       --------------  -----  
 0   Order_No                     7068 non-null   object 
 1   Time from Pickup to Arrival  3 non-null      float64
dtypes: float64(1), object(1)
memory usage: 110.6+ KB


In [16]:
Riders_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 960 entries, 0 to 959
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Rider Id        960 non-null    object 
 1   No_Of_Orders    960 non-null    int64  
 2   Age             960 non-null    int64  
 3   Average_Rating  960 non-null    float64
 4   No_of_Ratings   960 non-null    int64  
dtypes: float64(1), int64(3), object(1)
memory usage: 37.6+ KB
